### ISSUE TITLE
Non editable elements don't work as soon as they are replaced by new non editable elements (even when identical)

### ISSUE THREAD

**Issue by radiantshaw** [tinymce/tinymce#6557]
**What is the current behavior? Describe the bug**

1. Enable the `noneditable` plugin
2. Enter the following HTML through the "source code" modal: `<div id="non-editable-bug" class="mceNonEditable">Non editable bug</div>`
3. TinyMCE shouldn't allow to edit the "Non editable bug" text; which is expected
4. Now replace the entire `<div>` with the exact same contents using the following code:
   ```js
   tinymce.activeEditor.dom.setOuterHTML(tinymce.activeEditor.dom.select('#non-editable-bug'), '<div id="non-editable-bug" class="mceNonEditable">Non editable bug</div>')
   ```
5. Note that it now allows the "Non editable bug" text to be edited, which means the `<div>` is no longer non editable

**What is the expected behavior?**

The "Non editable bug" text given in the above steps shouldn't be allowed to edit after replacement since the `mceNonEditable` class is still present.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**

The issue is observed on TinyMCE `v5.2.1` on Edge. Didn't check it on other browsers and for previous versions. But I'm happy to do so if required.

**Additional notes**

The issue doesn't occur and it works as expected when the entire container element having the `mceNonEditable` class is not replaced but only the contents inside are replaced. That is:
```js
tinymce.activeEditor.dom.setHTML(tinymce.activeEditor.dom.select('#non-editable-bug'), 'Non editable bug')
```

**Comment by radiantshaw** [comment]
Here's also something which I found interesting. When you follow the steps given in the **What is the current behavior? Describe the bug** section in the description above, and after that you open the "source code" modal, and without doing anything, just hit "Save", the `<div>` becomes non editable again and works as expected.

**Comment by ashfordneil** [comment]
Hi @radiantshaw,

So the editor can only turn elements with the `mceNonEditable` class into non-editable nodes when the elements are inserted via the editor. When you edit HTML directly through the `dom` API (which is essentially just a wrapper around native APIs), you're responsible for making sure that the relevant attributes (in this case, `contenteditable="false"`) stay on the elements you're editing.

If you'd like to insert content in a way that the editor's filters (and sanitisers) will run on it, I'd recommend using something like the [editor.insertContent](https://www.tiny.cloud/docs/api/tinymce/tinymce.editor/#insertcontent) API*. This will insert content at the cursor location, but you can work around that using something like the following code.

```typescript
const bm = editor.selection.getBookmark();
editor.selection.setNode(...);
editor.insertContent(...);
editor.selection.moveToBookmark(bm);
```

 \* other examples include [editor.selection.setContent](https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.selection/#setcontent).

Also, the reason that the source code view will have that behaviour is that hitting "save" from there will call (among other things) `editor.setContent(editor.getContent())` and pass everything back through the filters and sanitisers again.

**Comment by radiantshaw** [comment]
Hello @ashfordneil,

I tried out the above solution given by you (albeit a bit differently based on the implementation), but sorry to say that it didn't work. I did the following:
```js
tinymce.activeEditor.selection.select(editor.dom.get('non-editable-bug'));
tinymce.activeEditor.selection.setContent('<div id="non-editable-bug" class="mceNonEditable">Non editable bug</div>')
```

**Comment by ashfordneil** [comment]
I made [this fiddle](https://fiddle.tiny.cloud/Z5haab) to try to reproduce the bug. I can reproduce it on Tiny 5.2.1, but by the looks of things it's something we fixed with the 5.3 release. Have a play with the fiddle on the latest stable release (which is 5.7 at the moment), and if you can find a way to reproduce the bug still then let us know.

**Comment by radiantshaw** [comment]
Thank you so much for providing the solution so quickly, @ashfordneil. It's working properly on the fiddle you provided. I need to check if it works in our app though. I'll revert accordingly when I get the chance.

BTW, if anyone from the future is wondering how to make it work in the versions before 5.3, you can add the `contenteditable="false"` attribute manually to the element you're trying to insert. As such:
```js
tinymce.activeEditor.selection.setContent('<div id="non-editable-bug" contenteditable="false">Non editable bug</div>')
```
And yes, if you're adding the `contenteditable` attribute manually, then you don't need to add the `mceNonEditable` class.
